src/OpenTelemetry.Api/Context/AsyncLocalRuntimeContextSlot.cs

```diff
@@ -24,13 +29,26 @@ public AsyncLocalRuntimeContextSlot(string name)
     }
 
     /// <inheritdoc/>
-    public object Value
+    public object? Value
     {
         get => this.slot.Value;
-        set => this.slot.Value = (T)value;
+        set
+        {
+            if (typeof(T).IsValueType && value is null)
+            {
+                this.slot.Value = default!;
+            }
+            else
+            {
+                this.slot.Value = (T)value!;
+            }
+        }
     }
 
     /// <inheritdoc/>
+#if NET
+    [return: MaybeNull]
+#endif
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public override T Get()
```

Comment:
I'm also interested in learning :)

Reply:
When working with generics, the return type often needs to be flexible to accommodate both value types and reference types.
If you specify T? as a return type in a generic method, that means you're indicating that T can be null. However, value types cannot be null unless they're wrapped in Nullable<T> (e.g. int?).
Using [return: MaybeNull] allows the return type to be non-nullable for value types (e.g. int) but still allows null to be returned for reference types (string). This provides more flexibility without requiring every caller to deal with nullable value types.

Reply:
Good question(s)! Had to do some research to really answer this 🤣 

Short answer: It looks like I'm a bit of a dinosaur when it comes to my nullable annotation style!

Long answer: 

So the way I did it here is the way you _had_ to do things when nullable annotations first dropped. `AllowNull` and `MaybeNull` ended up all over generic code. A really good example is [AsyncLocal&lt;T&gt;](https://github.com/dotnet/runtime/blob/e49d49711d809b3c1c1cb78fbc58233afe8d2ff1/src/libraries/System.Private.CoreLib/src/System/Threading/AsyncLocal.cs#L30-L47) itself. This is how I learned to do things.

It seems in C# 9 the language added [Unconstrained type parameter annotations](https://learn.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-9.0/unconstrained-type-parameter-annotations) which makes the more natural sylte (`T?`) @Kielek  is asking about possible.

What should we do? At the end of the day, both styles do the same thing. I'm happy to change it or leave as it just LMK what you guys would prefer. I'm leaning towards keeping `MaybeNull` because this class is a wrapper over `AsyncLocal<T>` and that's what it does, but I don't feel strongly about it.

Reply:
I don't mind the `MaybeNull`, I think it's easy to read and understand.

But if the Unconstrained type parameter annotation (`T?`) works without needing the preprocessor (`#if NET`) then that would be my vote.

Reply:
+1 for @TimothyMothra comment.
One more thing. PublicApi files ignores attributes, if possible I would go w C#9+ approach.

Reply:
Updated